### DISCUSSION TITLE
Graphql srcset does not except format argument in the same way url does as you'd expect.

### DISCUSSION THREAD

**Discussion by toddpadwick** [craftcms/cms#10299] [Q&A] [answered]
### Description
I need to create webp versions of my asset URLs and srcsets generated in graphql. adding format to URL `@transform` argument works as expected. However, it does not work in the `srcset` argument. There is very little documentation that i can find on the srcset in graphql in the craft docs to provide further information on this.

```
    image {
         title
         width
         height
         url @transform(width: 300)
         srcset(sizes: ["300","600", "960", "1200"])
         url @transform(width: 300, format:"webp")
         srcsetWebp:srcset(sizes: ["300","600", "960", "1200"], format:"webp")
    }
```

I have tried to do this, but even if this approach would work it would be long winded and require repeating my srcset multiple times... 

```
{ 
  imageWebp @transform(format: "webp") {
          title
         width
         height
         url @transform(width: 300) /// this does not work because the transform handle is applied above , however i only want the URL to be transformed to 300
         srcset(sizes: ["300","600", "960", "1200"])
  }

    image {
          title
         width
         height
         url @transform(width: 300)  // repeating myself
         srcset(sizes: ["300","600", "960", "1200"]) // repeating myself
  }

```

### Additional info
PHP version	8.0.12
OS version	Linux 4.15.0-50-generic
Database driver & version	MySQL 8.0.23
Image driver & version	Imagick 3.5.1 (ImageMagick 7.0.10-62)
Craft edition & version	Craft Pro 3.7.20
Yii version	2.0.43
Twig version	v2.14.8
Guzzle version	7.4.0

**Comment by andris-sevcenko** [comment]
> There is very little documentation that i can find on the srcset in graphql in the craft docs to provide further information on this.

It's meant to serve as a reference at this point, though, as GraphQL is self-documenting. You should be able to find all the available information using the built-int GraphiQL IDE. If you check there, you'll see the `srcset` field only has the `size` argument. It's, of course, not very obvious that you would be better off looking inside GraphiQL IDE and we're looking into improving the docs, too, so they serve better.

As for your specific case, you can get away with using

```graphql
{ 
  entries {
    ... on someEntryType {
      # This part is relevant
      imageWebp: assetField @transform(format: "webp") {
        ...defaultImageData
        url @transform(width: 300, format: "webp")
      }

      assetField {
        ...defaultImageData
        url @transform(width: 300)
      }
      # End relevant part
    }
  }
}

# Well the fragment is relevant, too
fragment defaultImageData on AssetInterface {
  title
  width
  height
  srcset(sizes: ["300","600", "960", "1200"])
}
```

Does that make sense?

**Reply by olsp** [reply]
It seems like this is no longer working in Craft 4. I tried it with a Craft 3 install and there the webp images are generated. On a Craft 4 install, I also get .jpg URLs in the "imageWebp".

**Reply by brandonkelly** [reply]
@olsp This may have been fixed by 32b05731a57. Please change your `craftcms/cms` requirement in composer.json to `dev-develop as 4.0.2` and run `composer update`, and let m know if you still experience the issue.

**Reply by olsp** [reply]
Seems like nothing changes regarding this issue when switching to dev-develop.

**Reply by brandonkelly** [reply]
Is it working for `imageWebp.url`? I would expect it to not work for `assetField.url` as its `@transform` directive doesn’t specify a format.

**Reply by olsp** [reply]
Yes, it does work for `imageWebp.url`. But in  `imageWebp.srcset`, there are no webp urls

**Reply by brandonkelly** [reply]
Fixed now for the next release, via #11413.

**Comment by brandonkelly** [comment, marked as answer]
Craft 4.0.5 is now out with the fix for this (#11413).
